static/js/containers/pages/CheckoutPage.js

```diff
@@ -202,16 +206,28 @@ export class CheckoutPage extends React.Component<Props, State> {
 
   render() {
     const { basket, requestPending } = this.props
-    const { errors } = this.state
+    const { errors, isLoading } = this.state
 
     const item = basket && basket.items[0]
     if (!basket || !item) {
       return (
         <DocumentTitle title={`${SETTINGS.site_name} | ${CHECKOUT_PAGE_TITLE}`}>
-          <div className="checkout-page">
-            No item in basket
-            {formatErrors(errors)}
-          </div>
+          {!isLoading ? (
+            <div className="checkout-page">
+              No item in basket
+              {formatErrors(errors)}
+            </div>
+          ) : (
+            <div className="checkout-page  checkout-loader text-center align-self-center">
+              <div className="loader-area">
+                <img
+                  src="/static/images/loader.gif"
+                  className="mx-auto d-block"
+                />
+                One moment while we prepare checkout
```

Comment:
does `while` here pointing to js keyword ?

Reply:
nope, its the element text.